src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathService.java

```diff
@@ -101,18 +107,28 @@ public LearningPathService(UserRepository userRepository, LearningPathRepository
         this.learningPathNgxService = learningPathNgxService;
         this.lectureUnitCompletionRepository = lectureUnitCompletionRepository;
         this.studentParticipationRepository = studentParticipationRepository;
+        this.learningPathsConfigurationRepository = learningPathsConfigurationRepository;
     }
 
     /**
      * Enable learning paths for the course and generate learning paths for all students enrolled in the course
      *
      * @param course course the learning paths are created for
+     * @return the learning paths configuration of the course
      */
-    public void enableLearningPathsForCourse(@NotNull Course course) {
+    public LearningPathsConfiguration enableLearningPathsForCourse(@NotNull Course course) {
         course.setLearningPathsEnabled(true);
+
+        if (course.getLearningPathsConfiguration() == null) {
+            course.setLearningPathsConfiguration(new LearningPathsConfiguration());
```

Comment:
is this for legacy data? Otherwise one could create a config on course creation?

Reply:
Since many courses currently do not use learning paths I opted to only create settings for them if they are actually needed